engine/src/test/java/io/camunda/zeebe/engine/processing/signal/SignalIntermediateThrowEventTest.java

```diff
@@ -174,44 +175,26 @@ public void shouldApplyOutputMappings() {
 
   @Test
   public void shouldNotSendResponseWhenPartOfBatch() {
-    // given
-    final String businessKey = "42";
-    final String messageName = "start";
+    clearInvocations(ENGINE.getCommandResponseWriter());
 
+    // given
     final var process =
         Bpmn.createExecutableProcess(PROCESS)
             .startEvent()
-            .intermediateCatchEvent(
-                "start",
-                b -> b.message(m -> m.name(messageName).zeebeCorrelationKey("=businessKey")))
             .intermediateThrowEvent("signal", b -> b.signal(SIGNAL_NAME_1))
             .endEvent()
             .done();
 
     ENGINE.deployment().withXmlResource(process).deploy();
 
-    final var processInstanceKey =
-        ENGINE
-            .processInstance()
-            .ofBpmnProcessId(PROCESS)
-            .withVariables(Map.of("businessKey", businessKey))
-            .create();
-
-    assertThat(
-            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
-                .withProcessInstanceKey(processInstanceKey)
-                .withElementId("start")
-                .limit(1)
-                .exists())
-        .isTrue();
-
     // when
-    ENGINE.message().withName(messageName).withCorrelationKey(businessKey).publish();
+    ENGINE.processInstance().ofBpmnProcessId(PROCESS).create();
 
     assertThat(
             RecordingExporter.signalRecords(SignalIntent.BROADCASTED)
                 .withSignalName(SIGNAL_NAME_1)
-                .limit(1)
+                // No need to use .limit(1) here as .exists() uses .findFirst() in order to
+                // determine if the element is present.
                 .exists())
```

Comment:
🔧 Feel free to remove this comment and introduce Javadoc on `.exists()`'s declaration, clarifying that it is a short-circuiting terminal operation. This would help all users of the exists method when they read it's javadoc